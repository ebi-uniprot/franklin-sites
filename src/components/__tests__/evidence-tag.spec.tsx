import { render, fireEvent, screen } from '@testing-library/react';

import EvidenceTag from '../evidence-tag';

describe('EvidenceTag component', () => {
  it('should render tag and toggle visibility of content', () => {
    const { asFragment } = render(
      <EvidenceTag label="My evidence" title="My title">
        <p>Some content</p>
      </EvidenceTag>
    );
    const contentNode = screen.getByTestId('evidence-tag-content');
    expect(contentNode).not.toHaveClass('evidence-tag-content--visible');
    const triggerNode = screen.getByTestId('evidence-tag-trigger');
    fireEvent.click(triggerNode);
    expect(contentNode).toHaveClass('evidence-tag-content--visible');
    expect(asFragment()).toMatchSnapshot();
  });
});
