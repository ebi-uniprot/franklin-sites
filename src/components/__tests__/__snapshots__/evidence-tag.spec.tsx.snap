// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceTag component should render tag and toggle visibility of content 1`] = `
<DocumentFragment>
  <button
    aria-controls="abcd"
    aria-expanded="true"
    class="evidence-tag"
    data-testid="evidence-tag-trigger"
    type="button"
  >
    <test-file-stub
      height="12"
      width="12"
    />
    <span
      class="evidence-tag__label"
    >
      My evidence
    </span>
  </button>
  <div
    class="evidence-tag-content evidence-tag-content--visible"
    data-testid="evidence-tag-content"
    id="abcd"
  >
    <p>
      Some content
    </p>
  </div>
</DocumentFragment>
`;
